packages/logging_cloudwatch/aws_logging_cloudwatch/lib/src/cloudwatch_logger_plugin.dart

```diff
@@ -301,11 +309,35 @@ class CloudWatchLoggerPlugin extends AWSLoggerPlugin
 
   /// Whether a [logEntry] should be logged by this plugin.
   bool _isLoggable(LogEntry logEntry) {
-    if (!_enabled) {
-      return false;
-    }
+    if (!_enabled) return false;
+
     final loggingConstraint = _getLoggingConstraint();
-    return logEntry.level >= loggingConstraint.defaultLogLevel;
+    final hasUserLogLevel = loggingConstraint.userLogLevel.containsKey(_userId);
+    LogLevel? logLevel;
+
+    if (hasUserLogLevel) {
+      final userLogLevel = loggingConstraint.userLogLevel[_userId]!;
+      logLevel =
+          _getCategoryLogLevel(userLogLevel.categoryLogLevel, logEntry) ??
+              userLogLevel.defaultLogLevel;
+    } else {
+      logLevel =
+          _getCategoryLogLevel(loggingConstraint.categoryLogLevel, logEntry);
+    }
+
+    return logEntry.level >= (logLevel ?? loggingConstraint.defaultLogLevel);
```

Comment:
if userLogLevel defaults to userLogLevel.defaultLogLevel, shouldn't we just default to defaultLogLevel for the !hasUserLogLevel case?

Reply:
If there is a categoryLogLevel for the category that the LogEntry is in then we would use that over the defaultLogLevel in the case of !hasUserLogLevel